30_regexp_lab/relab.txt

```diff
@@ -1,17 +1,19 @@
 Use grep to print only the lines from "input" for which:
 
 1. There is a binary number in the line
+	cat input |grep -a [0-9]
 
 2. There is an even binary number in the line
-
+	cat input |grep -a *[0-9]
 3. Line includes a hexadecimal number
-
+	cat input |grep '[xNN]'
```

Comment:
A hexa decimal number can either start in 0x or not, and in any case it is followed by normal digits. The regexp is therefore '(0x)?[0-9]+' which means the sequence 0x may appear (0 or 1 times), and after it a sequence of 1 or more normal digits